extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -413,6 +416,12 @@ export class AmpA4A extends AMP.BaseElement {
                 this.win.ampA4aValidationKeys = this.getKeyInfoSets_();
                 return this.verifyCreativeSignature_(
                     creativeParts.creative, creativeParts.signature);
+              })
+              // Emit adResponseValidateEnd for any non-erroneous validation
+              // completion, but only if we've actually started validation.
+              .then(creative => {
```

Comment:
Is there some cost to adding a promise (e.g. JS context switch)?  Instead you can move this within verifyCreativeSignature